Firebase/InAppMessaging/Flows/FIRIAMDisplayExecutor.m

```diff
@@ -307,14 +309,33 @@ - (void)displayMessageLoadError:(NSError *)error {
 
   UIAlertAction *defaultAction = [UIAlertAction actionWithTitle:@"OK"
                                                           style:UIAlertActionStyleDefault
-                                                        handler:^(UIAlertAction *action){
+                                                        handler:^(UIAlertAction *action) {
+                                                          self.alertWindow.hidden = NO;
+                                                          self.alertWindow = nil;
                                                         }];
 
   [alert addAction:defaultAction];
 
-  [[UIApplication sharedApplication].keyWindow.rootViewController presentViewController:alert
-                                                                               animated:YES
-                                                                             completion:nil];
+  dispatch_async(dispatch_get_main_queue(), ^{
+#if defined(__IPHONE_13_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+    if (@available(iOS 13.0, *)) {
```

Comment:
would this two conditions redundant?

Reply:
Yes, Xcode complains without the inner check, though.